### DISCUSSION TITLE
Ability to export all traces/sessions/scores from langfuse and import back to another langfuse instance

### DISCUSSION THREAD

**Discussion by SinghCoder** [langfuse/langfuse#3240] [Ideas]
### Describe the feature or potential improvement

In our tests setup, we run langfuse to visualize traces if something goes wrong with some test instance. To keep things reproducible, we dump langfuse traces as csv artifacts, but currently, we don't have a way to import it back to langfuse.
We can write a script to import it back in a new langfuse instance, but a native export/import functionality will take the experience to next level.

### Additional information

Full discussion in https://github.com/orgs/langfuse/discussions/1671#discussioncomment-10553762.

**Comment by marcklingen** [comment]
thanks for creating this thread, I am curious in the general interest in this. 

Quoting from the original [thread](https://github.com/orgs/langfuse/discussions/1671#discussioncomment-10553762.):

> You could just use the langfuse python/js sdk to fetch the data and then add it to another instance, thereby you do not have to rely on interfaces that might break while this is not a maintained feature
> Unfortunately I agree on this, this is not a small change. Thereby we will need to maintain a storage format that is forwards/backwards compatible. I think this is not worth the maintenance effort right now unless more teams are interested in this.

**Comment by dsaad68** [comment]
This is really important for organizations and enterprises, so they can create reports.

**Reply by marcklingen** [reply]
Yes. All the data is available via the api. Can you let me know which kind of export you’re missing? Happy to help

**Reply by Talha-Tahir-Khurshid** [reply]
A simple export button on the UI which exports the results of a Experiment run as a CSV would be very helpful.

**Reply by christopher-hoernle** [reply]
There was a button but now its gone again? (checking the demo)

**Reply by marcklingen** [reply]
You can export trace, sessions and generations right from the UI. Can you try in your own project?

**Reply by christopher-hoernle** [reply]
@marcklingen We are still running 2.X on Azure :(

**Reply by marcklingen** [reply]
Ok makes sense, what blocks you from upgrading to the latest version?

**Reply by christopher-hoernle** [reply]
@marcklingen Currently we are using [this project](https://github.com/Azure-Samples/langfuse-on-azure). I know its possible to deploy on Azure with Kubernetes but its too complex. If the azure sample doesn't get updated we will have to see.

**Comment by vmalyi** [comment]
But what about exporting sessions?

Doesn't seem to be available via the UI, or am I missing anything? 

<img width="1344" height="209" alt="Bildschirmfoto 2025-10-07 um 08 33 43" src="https://github.com/user-attachments/assets/02062220-ef84-464c-aaef-17926c51c13d" />

**Comment by jannikmaierhoefer** [comment]
Hi all, we published a Python script to migrate traces, prompts, sessions and scores from one Langfuse project to another one: https://langfuse.com/guides/cookbook/example_data_migration
